File: Mobile/mini_strava/lib/features/profile/domain/entities/user_profile.dart
enum Gender { male, female, other, notSet }

class UserProfile {
  final String firstName;
  final String lastName;
  final DateTime? birthDate;
  final Gender gender;
  final int? heightCm;
  final double? weightKg;


  final String? avatarPathOrUrl;

  const UserProfile({
    required this.firstName,
    required this.lastName,
    required this.birthDate,
    required this.gender,
    required this.heightCm,
    required this.weightKg,
    this.avatarPathOrUrl,
  });

  UserProfile copyWith({
    String? firstName,
    String? lastName,
    DateTime? birthDate,
    Gender? gender,
    int? heightCm,
    double? weightKg,
    String? avatarPathOrUrl,
  }) {
    return UserProfile(
      firstName: firstName ?? this.firstName,
      lastName: lastName ?? this.lastName,
      birthDate: birthDate ?? this.birthDate,
      gender: gender ?? this.gender,
      heightCm: heightCm ?? this.heightCm,
      weightKg: weightKg ?? this.weightKg,
      avatarPathOrUrl: avatarPathOrUrl ?? this.avatarPathOrUrl,
    );
  }
}

File: API/Strava_DreamTeam_Edition_API/Models/Domain/UserProfile.cs


using System.ComponentModel.DataAnnotations;

namespace Strava_DreamTeam_Edition_API.Models.Domain
{
    public class UserProfile
    {
        [Key]
        public string UserId { get; set; } = default!;

        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public DateTime? BirthDate { get; set; }


        public string? Gender { get; set; }

      
        public decimal? HeightCm { get; set; }
        public decimal? WeightKg { get; set; }

        public byte[]? Avatar { get; set; }
        public string? AvatarContentType { get; set; }

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}